astropy/coordinates/representation.py

```diff
@@ -421,39 +421,56 @@ class CartesianRepresentation(BaseRepresentation):
 
     Parameters
     ----------
-    x, y, z : `~astropy.units.Quantity`
-        The x, y, and z coordinates of the point(s). If ``x``, ``y``, and
-        ``z`` have different shapes, they should be broadcastable.
-
+    x, y, z : `~astropy.units.Quantity` or array
+        The x, y, and z coordinates of the point(s). If ``x``, ``y``, and ``z``
+        have different shapes, they should be broadcastable. If not quantity,
+        ``unit`` should be set.  If only ``x`` is given, it is assumed that it
+        contains an array with the 3 coordinates are stored along ``axis``.
+    unit : `~astropy.unit.Unit` or str
+        If given, the coordinates will be converted to this unit (or taken to
+        be in this unit if not given.
+    axis : int, optional
+        The axis along which the coordinates are stored for the case an array
+        of vectors is passed in (default: 0).
```

Comment:
maybe ``case an array of vectors is passed in`` -> ``when a single array is provided rather than distinct `x` `y`, and `z` ``?  To me, at least, that seems clearer in the context